sdk/azidentity/device_code_credential.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+const (
+	deviceCodeGrantType = "urn:ietf:params:oauth:grant-type:device_code"
+)
+
+// DeviceCodeResult is used to store device code related information to help the user login and allow the device code flow to continue
+// to request a token to authenticate a user
+type DeviceCodeResult struct {
```

Comment:
Does this actually need to be exported?  Isn't all of this implementation details for how device code polling works?

Reply:
Yes, that's what my original comment said. I don't think it needs to be exported.